### PR TITLE
Created database for subreddits

### LINKED ISSUE #12: Create database table for CollegiateSubreddits
Create database table for CollegiateSubreddits

# Acceptance Criteria:

- [x] There is an `@Entity` class called CollegeSubreddit.java
- [x] There is a `@Repository` class called CollegeSubredditRepository.java
- [x] When you start up the repo on localhost, you can see the table
      using the H2 console (see the file `docs/h2-database.md` for 
      instructions.)
- [x] When you deploy the app to Heroku, you can see the table
      using the psql command (See `Accessing Database Console` in the
      main `README.md` file for instructions.)

# Details: `@Entity` class

The `@Entity` class should be similar to the file [`Todo.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/entities/Todo.java)
in the example repo <https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2>.  

You may also want to consult `User.java` in the same repo for another example.

Your file will be called `CollegiateSubreddit.java` and should be placed
in the `entities` subdirectory of the repo.

It will have a field for `id`, which should be defined like this:

```
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;
```

And fields for `name`,`location`, and `subreddit`, each of which should be strings, like this:

```
  private String name;
  private String location;
  private String subreddit;
```

Also note that the line:

```
@Entity(name = "todos")
```

should be replaced with this, since Spring Boot requires `CamelCase` while SQL requires `snake_case`:

```
@Entity(name = "collegiate_subreddits")
```

The rest, you should be able to figure out on your own.

The main difference is that you do NOT need the following in front of
any of your attributes.  

```
@ManyToOne
@JoinColumn(name = "user_id")
```

This preceeds the attribute:

```
  private User user;
```

which is how we indicate that a particular record belongs to a particular `User`.  We aren't doing that in the case of the records in the `CollegeSubreddit` table, though it is a common pattern in many
applications.

# Details: `@Repository` class

For the repository class, see the example [`TodoRepository.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/repositories/TodoRepository.java).

You may also consult `UserRepository.java` in the same repo for another example.

Looking at the sample code, let's figure out what needs to be changed:

```
@Repository
public interface TodoRepository extends CrudRepository<Todo, Long> {
  Iterable<Todo> findAllByUserId(Long user_id);
}
```

1. You'll need to change the name `TodoRepository` to `CollegiateSubredditRepository`
2. You'll need to change `Todo` to `CollegiateSubreddit`.  This `Todo` refers back to the `@Entity` class for a single row in the database.
3. You'll need to replace the method:
   ```
   Iterable<Todo> findAllByUserId(Long user_id);
   ```

   With these:

   ```
   Iterable<CollegiateSubredddit> findByName(String name);
   Iterable<CollegiateSubredddit> findBySubreddit(String subreddit);
   ```

   Note that one of the suprising things about Spring Boot is that 
   you *do not need to write code for these methods*.  Instead, the
   part of the Spring Boot framework known as *Spring Data JDBC* 
   looks at the name you choose for these methods
   and based on that, it *writes them for you*. 

   The rules for translating method naming conventions into 
   generated code are complicated: we will not go over all of them in lecture, and you are not expected to memorize or learn them all.
   Instead, you need to be able to find what you need, when you need it.

   Some documentation is here to help get you started: <https://docs.spring.io/spring-data/jdbc/docs/current/reference/html/#jdbc.query-methods>

   While in this case, we are telling you what methods to add,
   in future cases, you'll be expected to figure out from context
   what kinds of lookup methods you might need for a table, consult
   the Spring Data

### PR DESCRIPTION
**Pull request by qianyouwang**
In this PR, the database for collegiatesubreddits is correctly setup. 

Below shows the ability to access the local H2 console.
Screen Shot 2022-02-03 at 17 28 04

Below shows the ability to access the remote heroku database console. 
Screen Shot 2022-02-03 at 17 30 27

Closes #12

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/repositories/CollegiateSubredditRepository.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package edu.ucsb.cs156.team02.repositories;
+
+import edu.ucsb.cs156.team02.entities.CollegiateSubreddit;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
```
Comment (SwiftWinds) [inline src/main/java/edu/ucsb/cs156/team02/repositories/CollegiateSubredditRepository.java]: Unneeded import
```suggestion
```

**Comment by SwiftWinds** [comment]
If you could, please link the issue in future PRs so reviewers can easily see what the acceptance criteria are. Thanks! :)

File: src/main/java/edu/ucsb/cs156/team02/entities/CollegiateSubreddit.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  // This establishes that many todos can belong to one user
+  // Only the user_id is stored in the table, and through it we
+  // can access the user's details
+
+  // @ManyToOne
+  // @JoinColumn(name = "user_id")
+  // private User user;
+  // private String title;
+  // private String details;
+  // private boolean done;
```
Comment (SwiftWinds) [inline src/main/java/edu/ucsb/cs156/team02/entities/CollegiateSubreddit.java]: Remove commented out lines
```suggestion
```

File: src/main/java/edu/ucsb/cs156/team02/entities/CollegiateSubreddit.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package edu.ucsb.cs156.team02.entities;
+
+
+import javax.persistence.Entity;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
```
Comment (SwiftWinds) [inline src/main/java/edu/ucsb/cs156/team02/entities/CollegiateSubreddit.java]: Remove unused imports
```suggestion
```

**Review by SwiftWinds — CHANGES_REQUESTED** [review]
Please take a look at my comments. Otherwise, lgtm!

**Comment by SwiftWinds** [comment]
(to approve all suggestions at once, go to the files tab and click add suggestion to batch on each one)

**Review by tekaireb — CHANGES_REQUESTED** [review]
I agree with the comments made by @SwiftWinds. Let me know when the unused imports have been removed so I can merge it.

**Review by SwiftWinds — APPROVED** [review]
thanks for the changes! Lgtm!

**Review by tekaireb — APPROVED** [review]

**Pull request merged**